src/lib/tvgShapeImpl.h

```diff
@@ -231,6 +237,7 @@ struct Shape::Impl
 
     void* update(RenderMethod& renderer, const RenderTransform* transform, uint32_t opacity, Array<RenderData>& clips, RenderUpdateFlag pFlag)
     {
+        this->renderer = &renderer;
```

Comment:
i think we need to call dispose() immediately when paints are popped from canvas.

Reply:
agree. I found simpler solution, please check it

Reply:
I left rdata = nullptr here to avoid crash

Reply:
@mihashco i undestood, need to figure out a better apporoach. let me think.